code-challenges.js

```diff
@@ -1,41 +1,265 @@
+    //  requiring a value of ' 0 ' to be added to the first number (' 1 '), resulting in the second
+    //  value also becomming equal to ' 1 '.
+    */
+
+    if ( nth > 1) {
+        out = fibonacci(--nth)
+        let [ a, b ] = [ out[out.length - 1], out[out.length - 2] ]
+        if( isNaN(b) )  { b = 0 }
+        return out.concat( a + b )
+    }
+    return out.concat(nth)
+}
+
+
+
+/// - - - - - - - - - - - - - - - -
+
+log(    "First 10 Digits from the Fibonacci Sequence",    
+        fibonacci(10)    
+    )
+
+/// - - - - - - - - - - - - - - - -
+
+
+
+
+
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
+
+// --------------------2)   Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
+
+var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]      // Expected output: [-9, 9, 7, 199]
+var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]     // Expected output: [-823, 7, 23]
+
+
+
+const sortOddNumbers = numArrIn => {
+    return (
+                numArrIn.filter( 
+                    v => typeof v === "number" 
+                ).filter( 
+                    v => Math.abs(v%2) === 1 
+                ).sort( (a,b) => a-b )
+           )
+}
+
+
+
+/// - - - - - - - - - - - - - - - -
+
+log(    "Sorting the Odd Numbers",    
+        
+        sortOddNumbers(fullArr1),    
+        "",
+        sortOddNumbers(fullArr2)    
+    )
+
+/// - - - - - - - - - - - - - - - -
+
+
+
+
+
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
+
+// --------------------3)   Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
+
+//  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /  / /
+
+var middleLetters1 = "hello"         // Expected output: “l”
+var middleLetters2 = "rhinoceros"    // Expected output: “oc”
+
+
+
+const getMiddleLetters = strIn => {
+    return (
+                strIn.length%2 === 0
+                ?
+                strIn.slice( (strIn.length / 2 - 1), (strIn.length / 2 + 1) )
+                :    
+                strIn.charAt( strIn.length / 2 - 0.5 )   
+           )
+}
+
+
+
+/// - - - - - - - - - - - - - - - -
+
+log(    "Middle Letter(s) of Words",   
+    
+        `\"${middleLetters1}\"        >    \' ${getMiddleLetters( middleLetters1 )} \'`,    
+        "",
+        `\"${middleLetters2}\"   >    \' ${getMiddleLetters( middleLetters2 )} \'`
+    )
```

Comment:
great job. and i really love the way that you handle your returns in terminal. very easy to read and straight-forward.